retrieval/manager.go

```diff
@@ -14,159 +14,133 @@
 package retrieval
 
 import (
-	"sync"
+	"fmt"
 
 	"github.com/go-kit/kit/log"
 	"github.com/go-kit/kit/log/level"
-	"golang.org/x/net/context"
-
 	"github.com/prometheus/prometheus/config"
-	"github.com/prometheus/prometheus/discovery"
 	"github.com/prometheus/prometheus/storage"
+	"golang.org/x/net/context"
 )
 
-// TargetManager maintains a set of targets, starts and stops their scraping and
-// creates the new targets based on the target groups it receives from various
-// target providers.
-type TargetManager struct {
-	append        Appendable
-	scrapeConfigs []*config.ScrapeConfig
-
-	mtx    sync.RWMutex
-	ctx    context.Context
-	cancel func()
-	wg     sync.WaitGroup
-
-	// Set of unqiue targets by scrape configuration.
-	targetSets map[string]*targetSet
-	logger     log.Logger
-}
-
-type targetSet struct {
-	ctx    context.Context
-	cancel func()
-
-	ts *discovery.TargetSet
-	sp *scrapePool
-}
-
+// Appendable enforced on the storage engine
 type Appendable interface {
 	Appender() (storage.Appender, error)
 }
 
-// NewTargetManager creates a new TargetManager.
-func NewTargetManager(app Appendable, logger log.Logger) *TargetManager {
-	return &TargetManager{
-		append:     app,
-		targetSets: map[string]*targetSet{},
-		logger:     logger,
+//
+func NewScrapeManager(ctx context.Context, logger log.Logger, app Appendable) *ScrapeManager {
+
+	return &ScrapeManager{
+		ctx:           ctx,
+		append:        app,
+		logger:        logger,
+		actionCh:      make(chan func()),
+		scrapeConfigs: make(map[string]*config.ScrapeConfig),
+		scrapePools: make(map[string]struct {
+			scrapePool *scrapePool
+			ctx        context.Context
```

Comment:
I'm having a hard time distinguishing the lifetime of this ctx vs scrapePool.ctx. I believe it's not needed here. Then we can just have the type be `map[string]*scrapePool`.

Reply:
thanks that simplifies it, I bit of confusion on my side.